pkg/auth/keymanager_test.go

```diff
@@ -217,3 +217,33 @@ func (s *TestKeyManagerSuite) TestKeyFetching() {
 		}
 	})
 }
+
+func (s *TestKeyManagerSuite) TestE2EKeyFetching() {
+	s.Run("retrieve key for e2e-tests environment", func() {
+		s.Config.GetViperInstance().Set("environment", "e2e-tests")
+		keyManager, err := auth.NewKeyManager(s.Config)
+		require.NoError(s.T(), err)
+		keys := authsupport.GetE2ETestPublicKey()
+
+		for _, key := range keys {
+			// check if the keys are parsed correctly.
+			_, err = keyManager.Key(key.KeyID)
+			require.NoError(s.T(), err)
+		}
+	})
+
+	s.Run("fail to retrieve e2e keys for unit test environment", func() {
+		s.Config.GetViperInstance().Set("environment", "unit-tests")
```

Comment:
We don't care much about unit-test env here.. Let's make sure e2e keys are not used in two usecases:
- Set env var explicitly to "prod", create a configuration and check the key manager.
- The env var is not set at all, so the default environment is used (which is currently "prod").

Reply:
i had set it to unit-tests at random since this block specifically checks for e2e-tests environment. everything else (including prod) should fall into the else clause: https://github.com/codeready-toolchain/registration-service/pull/63/files#diff-5f5c78f921deb86823e059dbb7069b73R55 however, i don't mind changing it to prod.. the outcome should be no different though..

sure, i can add a test for default as well..

Reply:
@alexeykazakov  also.. it looks like when you are running unit tests environment is already set to "unit-tests" see here: https://github.com/codeready-toolchain/registration-service/blob/master/test/testsuite.go#L24

Reply:
changed unit-tests to prod here: bb4a4c252d6dc87253a67d997b6356e0c2dedb5d

Reply:
fixed here: https://github.com/codeready-toolchain/registration-service/pull/63/files#diff-31b519707082aeb8272b75c7cf2460edR235